### PR TITLE
Latest changes

### PR DESCRIPTION
**Pull request by hafezus**
Pull request pls

### CODE REVIEW

File: smart_contracts/contracts/Dextr.sol [hunk 1]
```diff
@@ -0,0 +1,96 @@
+pragma solidity ^0.8.3;
+
+// import './Ownable.sol';
```
Comment (witherblock) [inline smart_contracts/contracts/Dextr.sol]: Remove commented imports if you aren't using them

File: smart_contracts/contracts/Dextr.sol [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+//ERC20
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+//SafeMath
+//import "github.com/ConsenSysMesh/openzeppelin-solidity/blob/master/contracts/math/SafeMath.sol";
```
Comment (witherblock) [inline smart_contracts/contracts/Dextr.sol]: Remove commented imports if you aren't using them

File: smart_contracts/contracts/Dextr.sol [hunk 3]
```diff
@@ -0,0 +1,96 @@
+pragma solidity ^0.8.3;
+
+// import './Ownable.sol';
+
+//ERC20
```
Comment (witherblock) [inline smart_contracts/contracts/Dextr.sol]: Pointless comment

File: smart_contracts/contracts/Dextr.sol [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+//SafeMath
```
Comment (witherblock) [inline smart_contracts/contracts/Dextr.sol]: Pointless comment

File: smart_contracts/contracts/Dextr.sol [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+//import "github.com/ConsenSysMesh/openzeppelin-solidity/blob/master/contracts/math/SafeMath.sol";
+
+/* 
+@Title: DexTr token contract
+@auther: Verde 8)
+@notice: contract for DexTr token -- transfer tokens (between user addresses or the contract itself)
+Todo: check if approve function changes
```
Comment (witherblock) [inline smart_contracts/contracts/Dextr.sol]: Why is there a TODO?

File: smart_contracts/contracts/Dextr.sol [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    
+    //-----------------------------overridden functions from ERC20 contract-----------------------------//
+    function transfer(address _account, uint256 _amount) public override returns (bool){
+        //require(balanceOf(_account) + _amount <= totalSupply() && _amount >= remaining, "Transfer failed: Transfer amount exceeds remaining tokens or exceeds circulating amount (released tokens + remaining + _amount > 750,000 tokens.");
+        require(releasedTokens + remaining <= 750_000, "Transfer failed: Transfer amount exceeds remaining tokens or attempting to transfer over total supply.");
+        _transfer(address(this), _account, _amount);
+        remaining-=_amount;
+        releasedTokens+=_amount;
+        return true;
+    }
```
Comment (witherblock) [inline smart_contracts/contracts/Dextr.sol]: Why are you doing this?

File: smart_contracts/contracts/Dextr.sol [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    function approve(address _spender, uint256 _amount) public override returns (bool) {
+        _approve(_msgSender(), _spender, _amount);
+        //remaining -= _amount;
+        releasedTokens+=_amount;
+        return true;
+    }
```
Comment (witherblock) [inline smart_contracts/contracts/Dextr.sol]: Unnecessary

File: smart_contracts/contracts/Dextr.sol [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+    //-----------------------------Dextr-----------------------------//
+    
+    //Mint DEX (must be less than 1 mil)
+    function mintDex(address _account, uint _amount) onlyOwner public { //change to private after testing
```
Comment (witherblock) [inline smart_contracts/contracts/Dextr.sol]: Has not been changed to private

File: smart_contracts/contracts/TradeDxt.sol [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+        //payable(address(this)).send(1000000000000 wei);
+        //_mint(address(this), 1000000 * (10 ** uint256(decimals())));
+    }
+    modifier nonZeroAddress(address _receiver) {
+        require(_receiver!=address(0), "Non-zero address check");
+        _;
+    }
+    function buyDxt(address _buyer, uint256 _amount) external payable {
+        //call from buyer's address
+        require(msg.sender == _buyer && _amount < totalSupply(), "Error: Not enough Ether in address"); // && _convertToEth(_amount) > msg.value
```
Comment (witherblock) [inline smart_contracts/contracts/TradeDxt.sol]: why require(msg.sender == _buyer) ?

If msg.sender is always going to be the buyer then why even ask the user to pass their own address?

**Review by witherblock — CHANGES_REQUESTED** [review]
I don't really understand what you are doing in the contracts. Didn't review the frontend, first get the contracts in order.

Notes:
- Write an explanation in the contracts on how they work, what each function does etc.
- Remove commented out lines if they are not going to be used.
- Use prettier to format files. The indent is using too many spaces which makes the code unreadable.
- There is no need of overriding transfer and approve functions. No clue why you did that.

File: smart_contracts/contracts/Dextr.sol (hunk 6 above)
Comment (hafezus) [reply smart_contracts/contracts/Dextr.sol]: For testing. Removed it now

File: smart_contracts/contracts/Dextr.sol (hunk 8 above)
Comment (hafezus) [reply smart_contracts/contracts/Dextr.sol]: Removed function altogether

**Pull request closed without merging**